tests/general/MementoBearTest.py

```diff
@@ -0,0 +1,306 @@
+import io
+import json
+import os
+import re
+import requests
+import requests_mock
+
+from bears.general.MementoBear import MementoBear
+
+from coalib.results.Result import Result
+from coalib.settings.Section import Section
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.testing.LocalBearTestHelper import LocalBearTestHelper
+
+from queue import Queue
+
+
+def get_testfile_path(name):
+    return os.path.join(os.path.dirname(__file__),
+                        'mementobear_test_files',
+                        name)
+
+
+def custom_matcher(request):
+    """
+    Mock the status codes for every request by taking the last three characters
+    of the request URL.
+
+    For connection checking url, it always passes 200 (prerequisite checking).
+
+    For URLs with no status codes appended, a ``RequestException`` is raised.
+
+    To test for URLs that redirect to URLs with a much larger length, the
+    ``redirect_long_url`` is returned.
+
+    :param request: The ``request`` that the mocker recieves.
+    :return:        A mocked ``Response`` object.
+    """
+    memento_timegate_pattern = re.compile(r'^http:\/\/timetravel\.mementoweb\.'
+                                          'org\/timegate\/.+')
+
+    # the connection check url needs to be explicitly
+    # set to 200
+    if request.url == MementoBear.check_connection_url:
+        status_code = 200
+    else:
+        try:
+            status_code = int(request.path_url[-3:])
+        except ValueError:
+            if memento_timegate_pattern.match(request.url):
+                # All memento timegate urls return 200
+                status_code = 200
+            else:
+                raise requests.exceptions.RequestException
+
+    resp = requests.Response()
+    resp.raw = io.BytesIO()
+    resp.request = request
+    resp.status_code = status_code
+    return resp
+
+
+# Load memento headers file
+with open(get_testfile_path('headers1.json')) as headers1:
+    with open(get_testfile_path('headers2.json')) as headers2:
+        memento_headers = [json.load(headers1),
+                           json.load(headers2)]
+
+
+def memento_archive_status_mock(m, url, is_archived=True, override_head=True):
+    """
+    Mock the memento result for the given url.
+
+    :param m:             A `request_mock.Mocker()` instance.
+    :param url:           The ``url`` to be mocked.
+    :param is_archived:   True means the url will be marked as
+                          archived.
+    :param override_head: True means the url's head request will be overriden
+                          by this method.
+    """
+    if override_head:
+        m.head(url, headers={})
+    if is_archived:
+        m.head(
```

Comment:
done in https://github.com/coala/coala-bears/pull/1719/commits/f524628017c96a46dba6560e250bfeb295407636 line 82